src/messageBroker.js

```diff
@@ -0,0 +1,45 @@
+const Pubsub = require('orbit-db-pubsub')
+const redis = require('redis')
+
+const createMessage = (heads, id) => JSON.stringify({ from: id, heads })
+const messageParse = (message) => JSON.parse(message)
+
+class MessageBroker extends Pubsub {
+  constructor (ipfs, id, instanceId, redisOpts, onMessageCallback) {
+    super(ipfs, id)
+    this._topics = {}
+    this.instanceId = instanceId
+    this.onMessageCallback = onMessageCallback
+    this.messageClientSub = redis.createClient(redisOpts)
+    this.messageClientPub = redis.createClient(redisOpts)
+    this.messageClientSub.on('message', this.messageHandler.bind(this))
+  }
+
+  async subscribe (topic, onMessageCallback, onNewPeerCallback) {
+    this.messageClientSub.subscribe(topic)
+    super.subscribe(topic, onMessageCallback, onNewPeerCallback)
+  }
+
+  async unsubscribe (topic) {
+    this.messageClientSub.unsubscribe(topic)
+    super.unsubscribe(topic)
+  }
+
+  messageHandler (topic, rawMessage) {
+    const message = messageParse(rawMessage)
+    if (message.from === this.instanceId) return
+    // console.log('On MESSAGE REDIS ---------')
+    // console.log(topic + ': ' + JSON.stringify(message))
+    this.onMessageCallback(topic, message.heads)
+    super.publish(topic, message.heads)
```

Comment:
this is the wrapped on message, so this is called only on redis pubsub messages, and publishes so that it is pushed to clients over libp2p. The original on message is called on libp2p messages. 

but this kind of logic could have problems in the future or be especially chatty if the peers are directly connected to each other. But right now there is network partition between any pinning node and its set of peers

Reply:
ok, makes sense then.